lib/inlines.js

```diff
@@ -834,7 +834,7 @@ var parseEntity = function(block) {
 };
 
 // Attempt to parse a url
-var reUrl = XRegExp.cache('^(?:[a-z][\\pL\\d-]+:(?:\\/{1,3}|[\\pL\\d%])|www\\d{0,3}[.]|[\\pL\\d.\\-]+[.]\\pL{2,4}\\/)(?:\\[[\\da-f:]+\\]|[^\\s`!()\\[\\]{};:\'",<>?«»“”‘’*_]|[*_]+(?=[^_*\\s])|[`!\\[\\]{};:\'",<>?«»“”‘’](?=[^\\s()<>])|\\((?:[^\\s()<>]|(?:\\([^\\s()<>]+\\)))*\\))+', 'i');
+var reUrl = XRegExp.cache('^(?:[a-z][\\pL\\d-]+([+.][a-z\\d]+)?:(?:\\/{1,3}|[\\pL\\d%])|www\\d{0,3}[.]|[\\pL\\d.\\-]+[.]\\pL{2,4}\\/)(?:\\[[\\da-f:]+\\]|[^\\s`!()\\[\\]{};:\'",<>?«»“”‘’*_]|[*_]+(?=[^_*\\s])|[`!\\[\\]{};:\'",<>?«»“”‘’](?=[^\\s()<>])|\\((?:[^\\s()<>]|(?:\\([^\\s()<>]+\\)))*\\))+', 'i');
```

Comment:
See comments on https://github.com/mattermost/commonmark.js/pull/7/files, some of which apply here as well :)

Reply:
Thanks for the review I checked the comment on the marked repo.